packages/interface/README.md

```diff
@@ -0,0 +1,77 @@
+# (Experimental) Interface
+
+The Interface Package contains the basis the start a modern WordPress screen as "core/edit-post" and "core/edit-site". The package offers a store and a set of components. The store is useful to contain common data required by a screen (e.g., active areas). The information is persisted across screen reloads. The components allow one to implement functionality like a sidebar and allow a screen to be extended by third-party plugins by default.
+
+
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/interface --save
+```
+
+_This package assumes that your code will run in an **ES2015+** environment. If you're using an environment that has limited or no support for ES2015+ such as lower versions of IE then using [core-js](https://github.com/zloirock/core-js) or [@babel/polyfill](https://babeljs.io/docs/en/next/babel-polyfill) will add support for these methods. Learn more about it in [Babel docs](https://babeljs.io/docs/en/next/caveats)._
+
+
+### API Usage
+
+
+
+#### Store
+
+#### Single Active Area Functionality
```

Comment:
I'm not sure i understand why we need separate actions/selectors for multiple or single active areas. I feel this could be just a prop of the `ComplementaryArea.Slot` component.

Reply:
Also, do we need both use-cases in the current screen?

Reply:
Hi @youknowriad, the single reducer handles the case where only one or zero areas are active in a given scope, so enabling an area disables the previous active are. It is used to track which complementary area (sidebar) is active in a given screen. The multiple reducer handles cases, where multiple areas can be active in a given scope, enabling an area, does not impacts the other areas. It is used to track which pinned items are shown.

Reply:
Yes, I understand buy why we need separate actions, can we just use the same actions (like we do for selected blocks for instance and their actions). The "prop" for the slot could be responsible of adapting the behavior properly.

What I'm saying is that a consumer shouldn't have to care about whether a single or multiple areas can be enabled at the same time, this should be internal to the Slot. The consumer should just enable/disable areas.

Reply:
Hi @youknowriad, so essentially the idea is to remove logic from the store the store? The store would just contain active areas and would be UP for the store users to have the logic and know if multiple active areas can be enabled or not?


If that's the case there are some disadvantages we would for example not be able to open a sidebar by just dispatching an action. In order to open a sidebar, one would need first to check if there is an active sidebar if yes dispatch an action to disable that area and then dispatch an action to enable the new sidebar.

Reply:
> so essentially the idea is to remove logic from the store the store? The store would just contain active areas and would be UP for the store users to have the logic and know if multiple active areas can be enabled or not?

Not really

> If that's the case there are some disadvantages we would for example not be able to open a sidebar by just dispatching an action

this should be a single action still but since the complementary slot has a config saying that it's a single active slot. It will automatically do its magic to disable the old one.

What I'm saying is that a consumer should do just `activateComplementaryArea( name )` (random naming here) regardless of whether the slot will display one or multiple at the same time. It's the responsibility of the implementation to know how it's configured, it shouldn't be the responsibility of the caller because otherwise for the same slot, some folks will say it's single while others will say it's multi